### ISSUE TITLE
incompatible requirements

### ISSUE THREAD

**Issue by dimbleby** [scipy/oldest-supported-numpy#70]
https://github.com/scipy/oldest-supported-numpy/blob/b0b3376f6838abd91a9988b936f70939d4acb775/setup.cfg#L45 and https://github.com/scipy/oldest-supported-numpy/blob/b0b3376f6838abd91a9988b936f70939d4acb775/setup.cfg#L70

would both apply for users of pypy on macos, but require different versions of numpy.  There may be other examples.

**Comment by rgommers** [comment]
Hmm, this is tested, but the tests say that PyPy is only supported on x86-64 and aarch64:

https://github.com/scipy/oldest-supported-numpy/blob/b0b3376f6838abd91a9988b936f70939d4acb775/tests/test_dependencies.py#L52

macOS uses `arm64` rather than `aarch64`. I'm not sure if this was an oversight in the test or PyPy is indeed not supported. I'm guessing the former, and we need to fix the metadata and add `arm64` to that line.

Cc @mattip as our PyPy expert.

**Comment by dimbleby** [comment]
Another pair that are at least logically incompatible - though possibly not in practice - is https://github.com/scipy/oldest-supported-numpy/blob/b0b3376f6838abd91a9988b936f70939d4acb775/setup.cfg#L54 and https://github.com/scipy/oldest-supported-numpy/blob/b0b3376f6838abd91a9988b936f70939d4acb775/setup.cfg#L32

I was experimenting with the use of this package via poetry, which is trying to build cross-platform solutions.  So it can't rely on out-of-band knowledge about which combinations _actually_ exist in the real world; but it may be that for an installer that only cares about the current platform these conflicts are never realised.

**Comment by rgommers** [comment]
Does Poetry actually error out on this? If so, I'm curious how and what "build cross-platform solutions" means exactly.

It's very difficult to make sure this is all logically consistent; the best way is to extend the platform/OS/interpreter combos being tested.

**Comment by dimbleby** [comment]
> Does Poetry actually error out on this?

Well it will do, once it has been fixed to behave properly!  I chanced across this project and thought - rightly, it turns out - that it would be an interesting testcase for some ongoing work in poetry to correct the handling of overlapping markers...

"build cross-platform solutions" means that poetry is supposed to build a portable lockfile.  ie installation from this lockfile might take place on any arbitrary platform, and poetry promises that regardless of the environment at install-time, it will be able to satisfy the top-level requirements using the details it has written in the lockfile.

Of course that's impossible if there are environments which have inconsistent requirements: there can never be any solution for those environments.
